ambry-frontend/src/test/java/com.github.ambry.frontend/AmbryBlobStorageServiceTest.java

```diff
@@ -510,58 +510,78 @@ public void injectionAccountAndContainerForPutTest() throws Exception {
+    blobId = new BlobId(BlobId.DEFAULT_FLAG, ClusterMapUtils.UNKNOWN_DATACENTER_ID, Account.UNKNOWN_ACCOUNT_ID,
         Container.UNKNOWN_CONTAINER_ID, clusterMap.getWritablePartitionIds().get(0)).getID();
     verifyAccountAndContainerFromBlobId(blobId, Account.UNKNOWN_ACCOUNT, Container.UNKNOWN_CONTAINER,
         RestServiceErrorCode.NotFound);
 
     // aid=unknownAId, cid=nonExistCId
-    blobId = new BlobIdV2(BlobId.DEFAULT_FLAG, ClusterMapUtils.UNKNOWN_DATACENTER_ID, Account.UNKNOWN_ACCOUNT_ID,
+    blobId = new BlobId(BlobId.DEFAULT_FLAG, ClusterMapUtils.UNKNOWN_DATACENTER_ID, Account.UNKNOWN_ACCOUNT_ID,
         (short) -1234, clusterMap.getWritablePartitionIds().get(0)).getID();
     verifyAccountAndContainerFromBlobId(blobId, null, null, RestServiceErrorCode.InvalidContainer);
 
     // aid=nonExistAId, cid=refCId
-    blobId =
-        new BlobIdV2(BlobId.DEFAULT_FLAG, ClusterMapUtils.UNKNOWN_DATACENTER_ID, (short) -1234, refContainer.getId(),
-            clusterMap.getWritablePartitionIds().get(0)).getID();
+    blobId = new BlobId(BlobId.DEFAULT_FLAG, ClusterMapUtils.UNKNOWN_DATACENTER_ID, (short) -1234, refContainer.getId(),
+        clusterMap.getWritablePartitionIds().get(0)).getID();
     verifyAccountAndContainerFromBlobId(blobId, null, null, RestServiceErrorCode.InvalidAccount);
 
     // aid=nonExistAId, cid=unknownCId
-    blobId = new BlobIdV2(BlobId.DEFAULT_FLAG, ClusterMapUtils.UNKNOWN_DATACENTER_ID, (short) -1234,
+    blobId = new BlobId(BlobId.DEFAULT_FLAG, ClusterMapUtils.UNKNOWN_DATACENTER_ID, (short) -1234,
         Container.UNKNOWN_CONTAINER_ID, clusterMap.getWritablePartitionIds().get(0)).getID();
     verifyAccountAndContainerFromBlobId(blobId, null, null, RestServiceErrorCode.InvalidAccount);
 
     // aid=nonExistAId, cid=nonExistCId
-    blobId = new BlobIdV2(BlobId.DEFAULT_FLAG, ClusterMapUtils.UNKNOWN_DATACENTER_ID, (short) -1234, (short) -11,
+    blobId = new BlobId(BlobId.DEFAULT_FLAG, ClusterMapUtils.UNKNOWN_DATACENTER_ID, (short) -1234, (short) -11,
         clusterMap.getWritablePartitionIds().get(0)).getID();
     verifyAccountAndContainerFromBlobId(blobId, null, null, RestServiceErrorCode.InvalidAccount);
   }
 
   /**
+   * Tests injecting target {@link Account} and {@link Container} for GET/HEAD/DELETE blobId string in {@link BlobId#BLOB_ID_V1}.
+   * The {@link AccountService} is prepopulated with a reference account and {@link Account#UNKNOWN_ACCOUNT}. The expected
+   * behavior should be:
+   * <pre>
+   *   AId in blobId    CId in blobId     expected Error      injected account      injected container
+   *    UNKNOWN           UNKNOWN          NotFound            UNKNOWN               UNKNOWN            This can succeed if the blob exists in backend.
+   * </pre>
+   * @throws Exception
+   */
+  @Test
+  public void injectionAccountAndContainerForGetHeadDeleteBlobIdV1Test() throws Exception {
```

Comment:
Is this test necessary? Do you need to check anymore how BlobId V1 treats account and container ids passed to it as this path will not be exercised anymore, would it?

Reply:
+1

Reply:
This is to test how `AmbryBlobStorageService` handle blobId string in v1.